From 519314d488624885a73f6204e3113aab850a36a0 Mon Sep 17 00:00:00 2001
From: Simon Dawson <spdawson@gmail.com>
Date: Wed, 3 Sep 2014 18:39:22 +0100
Subject: [PATCH] Complete wrapping of ntpshm_latch function in #ifdef
 NTPSHM_ENABLE
To: gpsd-dev@nongnu.org

Currently, the build fails as follows

gpsmon.o: In function `gpsmon_hook':
gpsmon.c:(.text+0x8b0): undefined reference to `ntpshm_latch'
collect2: error: ld returned 1 exit status
scons: *** [gpsmon] Error 1

Signed-off-by: Simon Dawson <spdawson@gmail.com>
---
diff -Nurp a/gpsd.h-tail b/gpsd.h-tail
--- a/gpsd.h-tail	2014-08-22 21:35:56.000000000 +0100
+++ b/gpsd.h-tail	2014-09-03 18:52:04.048627518 +0100
@@ -842,7 +842,9 @@ extern unsigned int ais_binary_encode(st
 extern void ntpshm_context_init(struct gps_context_t *);
 extern void ntpshm_session_init(struct gps_device_t *);
 extern int ntpshm_put(struct gps_device_t *, int, struct timedrift_t *);
+#ifdef NTPSHM_ENABLE
 extern void ntpshm_latch(struct gps_device_t *device,  /*@out@*/struct timedrift_t *td);
+#endif /* NTPSHM_ENABLE */
 extern void ntpshm_link_deactivate(struct gps_device_t *);
 extern void ntpshm_link_activate(struct gps_device_t *);
 
diff -Nurp a/gpsmon.c b/gpsmon.c
--- a/gpsmon.c	2014-08-22 21:06:58.000000000 +0100
+++ b/gpsmon.c	2014-09-03 18:52:23.172628059 +0100
@@ -778,8 +778,10 @@ static void gpsmon_hook(struct gps_devic
 
     report_unlock();
 
+#ifdef NTPSHM_ENABLE
     /* Update the last fix time seen for PPS. FIXME: do this here? */
     ntpshm_latch(device, &td);
+#endif /* NTPSHM_ENABLE */
 }
 /*@+observertrans +nullpass +globstate +compdef +uniondef@*/
 
